### ISSUE TITLE
Repeated `@ExtendWith` meta-annotations on fields no longer supported in 5.11.x

### ISSUE THREAD

**Issue by ppkarwasz** [junit-team/junit5#4054]
Since version 5.11.x, if a field is meta-annotated with multiple `@ExtendWith` annotations, JUnit fails to detect them.

The problem seems to be in `ExtensionUtils#streamExtensionFilteringFields`, which does not take into account repeatable annotations:

https://github.com/junit-team/junit5/blob/8e9094d60fa7964e635c6f4e91b7241266b3f09e/junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/ExtensionUtils.java#L204-L209

## Steps to reproduce

1. Meta-annotate some test class fields with multiple `@ExtendWith` annotations:

   ```java
   @Retention(RetentionPolicy.RUNTIME)
   @ExtendWith(Extension1.class)
   @ExtendWith(Extension2.class)
   public @interface ComposedAnnotation {}
   ```

   ```java
   public class TestClassWithFields {

     @ComposedAnnotation
     private static int staticFieldMetaAnnotatedTwice;

     @ComposedAnnotation
     private int instanceFieldMetaAnnotatedTwice;
   }
   ```

2. Run the following test cases: 

   ```java
   class ExtensionsUtilsTests {
     @Test
     void discoverExtensionsOnStaticFields() {
       ExtensionRegistrar registrar = mock();
       ExtensionUtils.registerExtensionsFromStaticFields(registrar, ClassWithFields.class);
       verify(registrar).registerExtension(Extension1.class);
       verify(registrar).registerExtension(Extension2.class);
     }
     @Test
     void discoverExtensionsOnInstanceFields() {
       ExtensionRegistrar registrar = mock();
       ExtensionUtils.registerExtensionsFromInstanceFields(registrar, ClassWithFields.class);
       verify(registrar).registerExtension(Extension1.class);
       verify(registrar).registerExtension(Extension2.class);
     }
   }
   ```

## Context

 - Used versions (Jupiter/Vintage/Platform): Jupiter 5.11.1 / Platform 1.11.1
 - Build Tool/IDE: Maven

**Comment by sbrannen** [comment]
Good catch, @ppkarwasz! 👍 

> The problem seems to be in `ExtensionUtils#streamExtensionFilteringFields`, which does not take into account repeatable annotations:

Indeed, the switch to the use of `isAnnotated()` there seems to be the culprit.

We'll work on a fix.

Cheers,

Sam

**Comment by marcphilipp** [comment]
This was introduced in #3895

**Comment by sbrannen** [comment]
I've reproduced this in the following standalone test class.

```java
package org.junit.jupiter.engine.descriptor;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.junit.jupiter.api.extension.Extension;
import org.junit.jupiter.engine.extension.ExtensionRegistrar;

class ExtensionsUtilsTests {

	@Test
	void discoverExtensionsOnStaticFields() {
		ExtensionRegistrar registrar = mock();
		ExtensionUtils.registerExtensionsFromStaticFields(registrar, TestClassWithFields.class);
		verify(registrar).registerExtension(Extension1.class);
		verify(registrar).registerExtension(Extension2.class);
	}

	@Test
	void discoverExtensionsOnInstanceFields() {
		ExtensionRegistrar registrar = mock();
		ExtensionUtils.registerExtensionsFromInstanceFields(registrar, TestClassWithFields.class);
		verify(registrar).registerExtension(Extension1.class);
		verify(registrar).registerExtension(Extension2.class);
	}

	static class Extension1 implements Extension {
	}

	static class Extension2 implements Extension {
	}

	@Retention(RetentionPolicy.RUNTIME)
	@ExtendWith(Extension1.class)
	@ExtendWith(Extension2.class)
	@interface MetaAnnotation {
	}

	static class TestClassWithFields {

		@MetaAnnotation
		static int staticFieldMetaAnnotatedTwice;

		@MetaAnnotation
		int instanceFieldMetaAnnotatedTwice;

	}

}
```

**Comment by sbrannen** [comment]
This has been fixed on `main` and backported to `5.11.x` for inclusion in 5.11.3.

**Comment by ppkarwasz** [comment]
> This has been fixed on `main` and backported to `5.11.x` for inclusion in 5.11.3.

Wow, this was fast! Thank You!
